src/Compilers/VisualBasic/Portable/VisualBasicParseOptions.vb

```diff
@@ -318,6 +239,32 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End Get
         End Property
 
+
+        Friend Overrides Sub ValidateOptions(builder As ArrayBuilder(Of Diagnostic))
+            ValidateOptions(builder, MessageProvider.Instance)
+
+            ' Validate LanguageVersion Not SpecifiedLanguageVersion, after Latest/Default has been converted
+            If Not LanguageVersion.IsValid Then
+                builder.Add(Diagnostic.Create(MessageProvider.Instance, ERRID.ERR_BadLanguageVersion, LanguageVersion.ToString))
+            End If
+
+            If Not PreprocessorSymbols.IsDefaultOrEmpty Then
+                For Each symbol In PreprocessorSymbols
+                    If Not IsValidIdentifier(symbol.Key) OrElse SyntaxFacts.GetKeywordKind(symbol.Key) <> SyntaxKind.None Then
+                        builder.Add(Diagnostic.Create(MessageProvider.Instance, ERRID.ERR_ProjectCCError1, VBResources.ERR_ExpectedIdentifier, symbol.Key))
+                    Else
+
+                        Debug.Assert(SyntaxFactory.ParseTokens(symbol.Key).Select(Function(t) t.Kind).SequenceEqual({SyntaxKind.IdentifierToken, SyntaxKind.EndOfFileToken}))
+
+                        Dim constant = InternalSyntax.CConst.TryCreate(symbol.Value)
+                        If constant Is Nothing Then
+                            builder.Add(Diagnostic.Create(MessageProvider.Instance, ERRID.IDS_InvalidPreprocessorConstantType, symbol.Key, symbol.Value.GetType))
```

Comment:
>IDS_InvalidPreprocessorConstantType [](start = 90, length = 35)

IDS_InvalidPreprocessorConstantType is neither an error number, nor a warning number, it simply designates a string in the resource file. It should not be used to create diagnostics.